azure/services/aso/interfaces.go

```diff
@@ -28,3 +29,11 @@ type Reconciler interface {
 	CreateOrUpdateResource(ctx context.Context, spec azure.ASOResourceSpecGetter, serviceName string) (result genruntime.MetaObject, err error)
 	DeleteResource(ctx context.Context, spec azure.ASOResourceSpecGetter, serviceName string) (err error)
 }
+
+// TagsGetterSetter represents an object that supports tags.
+type TagsGetterSetter interface {
+	GetAdditionalTags() infrav1.Tags
+	GetDesiredTags(resource genruntime.MetaObject) infrav1.Tags
+	GetActualTags(resource genruntime.MetaObject) infrav1.Tags
+	SetTags(resource genruntime.MetaObject, tags infrav1.Tags)
+}
```

Comment:
example showing how this is implemented: https://github.com/kubernetes-sigs/cluster-api-provider-azure/commit/5cba9bce64862e1dd2df83c1e5be7b3d60138808#diff-ea4ce1a96b7c0590d9ae02713171104af2511a49bc2fad674be02da8f69d5a6e

Reply:
Added the linked changes to this PR since I don't think we can reasonably merge tags separately from the rest of the ASO resource.